### PR TITLE
Improved tree-sitter queries

### PR DESCRIPTION
**Pull request by kirawi**
`@variable.property` -> `@variable.other.member`
`@property` -> `@variable.other.member`
`@number` -> `@constant.numeric.{integer, float}`
`@escape` -> `@constant.character.escape`
`@constant.character` is now uniformly used for all chars.

### CODE REVIEW

**Comment by kirawi** [comment]
@raygervais I noticed that
```toml
"ui.popup" = { bg = "base01" }
"ui.window" = { bg = "base00" }
"ui.help" = { bg = "base01", fg = "base06" }
```
have duplicated keys in `base16`.

**Comment by kirawi** [comment]
How do I fix the submodule change? I don't know how that even happened.

File: runtime/queries/json/highlights.scm [hunk 1]
```diff
@@ -4,6 +4,6 @@
 (string) @string
 
 (object
-  "{" @escape
+  "{" @constant.character.escape
   (_)
-  "}" @escape)
+  "}" @constant.character.escape)
```
Comment (kirawi) [inline runtime/queries/json/highlights.scm:9]: Should these be `@punctuation.bracket`?

**Comment by sudormrfbin** [comment]
> How do I fix the submodule change? I don't know how that even happened.

Is it the change with elixir ? #875

**Comment by kirawi** [comment]
It's with cpp, presumably this happened when I did `git pull upstream master`

File: runtime/queries/toml/highlights.scm [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-(boolean) @constant.builtin
+(boolean) @constant.builtin.boolean
 (comment) @comment
 (string) @string
-(integer) @number
-(float) @number
+(integer) @constant.numeric.integer
+(float) @constant.numeric.float
 (offset_date_time) @string.special
 (local_date_time) @string.special
```
Comment (kirawi) [inline runtime/queries/toml/highlights.scm:16]: We don't have a `@string.special` scope. I'm thinking this PR can become a general pass over all queries and scopes as well.

File: runtime/queries/yaml/highlights.scm [hunk 3]
```diff
@@ -1,12 +1,12 @@
-(block_mapping_pair key: (_) @property)
-(flow_mapping (_ key: (_) @property))
+(block_mapping_pair key: (_) @variable.other.member)
+(flow_mapping (_ key: (_) @variable.other.member))
 (boolean_scalar) @constant.builtin.boolean
 (null_scalar) @constant.builtin
 (double_quote_scalar) @string
 (single_quote_scalar) @string
 (escape_sequence) @string.escape
```
Comment (kirawi) [inline runtime/queries/yaml/highlights.scm:7]: TODO

File: runtime/queries/toml/highlights.scm (hunk 2 above)
Comment (kirawi) [reply runtime/queries/toml/highlights.scm:16]: NVM we do.

**Comment by archseer** [comment]
`git sync; git submodule update --init`?

File: runtime/queries/json/highlights.scm (hunk 1 above)
Comment (archseer) [reply runtime/queries/json/highlights.scm:9]: Yep. It's really barebones too, let's add number and boolean highlighting 

https://github.com/nvim-treesitter/nvim-treesitter/blob/bd1399105026c7b067dc197b31ccdb2602700a86/queries/json/highlights.scm

**Comment by kirawi** [comment]
Doesn't seem like it. Maybe my case is different since I set `upstream` to this repo while `origin` points to my fork?

**Comment by archseer** [comment]
Well now that you committed and pushed the submodule update it considers that to be what it's syncing to maybe?

**Comment by archseer** [comment]
Ah sorry, was supposed to be `git submodule sync` too. Easiest fix would be to manually update the submodule to the same commit as in the repo

File: runtime/queries/rust/highlights.scm [hunk 4]
```diff
@@ -81,9 +81,21 @@
...
 
 ; ---
-; Parameters
+; Variables
 ; ---
 
+(let_declaration
+  pattern: [
+    ((identifier) @variable)
+    ((tuple_pattern
```
[the remark this answers is not in the document]
Comment (archseer) [reply runtime/queries/rust/highlights.scm:88]: Not sure, we could do that and simplify

**Comment by kirawi** [comment]
How do you do that?

**Comment by kirawi** [comment]
Curse you git... curse you.

File: runtime/queries/json/highlights.scm [hunk 5]
```diff
@@ -1,9 +1,20 @@
+[
+  (true)
+  (false)
+] @constant.builtin.boolean
+(null) @constant.builtin
+(number) @constant.numeric
```
Comment (archseer) [inline runtime/queries/json/highlights.scm:6]: Maybe `constant.numeric.integer`? I know it's not necessarily correct, but if someone defines `constant.numeric.integer` and `constant.numeric.float` but not `constant.numeric` then this won't get highlighted.

Comment (archseer) [reply runtime/queries/json/highlights.scm:6]: Though I guess we should just recommend styling `constant.numeric`

File: runtime/themes/dark_plus.toml [hunk 6]
```diff
@@ -35,8 +35,8 @@
 
 "string" = { fg = "#ce9178" }
 "string.regexp" = { fg = "regex" }
-"number" = { fg = "#b5cea8" }
-"escape" = { fg = "#d7ba7d" }
+"constant.numeric" = { fg = "#b5cea8" }
+"constant.character" = { fg = "#d7ba7d" }
```
Comment (archseer) [inline runtime/themes/dark_plus.toml:39]: `constant.character.escape`?

Comment (kirawi) [reply runtime/themes/dark_plus.toml:39]: I believe that they both get styled the same, or at least they used to be.

**Pull request merged**